float32 right_forward
float32 right_side
float32 left_side
float32 left_forward
float32 sum_all
float32 sum_forward
